MdePkg/Library/StackCheckLib/X64/StackCheckFunctionsMsvc.nasm

```diff
@@ -0,0 +1,45 @@
+;------------------------------------------------------------------------------
+; X64/StackCheckFunctionsMsvc.nasm
+;------------------------------------------------------------------------------
+
+    DEFAULT REL
+    SECTION .text
+
+extern ASM_PFX(StackCheckFailure)
+extern ASM_PFX(__security_cookie)
+extern ASM_PFX(CpuDeadLoop)
+
+; Called when a buffer check fails. This functionality is dependent on MSVC
+; C runtime libraries and so is unsupported in UEFI.
+global ASM_PFX(__report_rangecheckfailure)
+ASM_PFX(__report_rangecheckfailure):
+    call ASM_PFX(CpuDeadLoop)

```

Comment:
What is the difference between this and `jmp $`?

Reply:
I don't think they are functionally different in this case because a return is the next instruction anyways. Do you think jmp is preferable?

Reply:
Went ahead and updated this to a jmp